File: Desafios em C#/1118 - Várias Notas Com Validação.cs
using System; 
using System.Globalization;

//Developed by: @LucasMarcuzo
class URI1118 {

    static void Main(string[] args) { 

        double m, n, t, calc=1;

        do
        {
            do
            {
                m=double.Parse(Console.ReadLine(),CultureInfo.InvariantCulture);

                if(m < 0.0f || m > 10.0f)
                {
                    Console.WriteLine("nota invalida");
                }
            
            } while (m < 0.0f || m > 10.0f);

            do
            {
                n=double.Parse(Console.ReadLine(),CultureInfo.InvariantCulture);

                if(n < 0.0f || n > 10.0f)
                {
                    Console.WriteLine("nota invalida");
                }
    
            } while (n < 0.0f || n > 10.0f);


            Console.WriteLine("media = {0:0.00}",t=(m+n)/2);

            do
            {
                Console.WriteLine("novo calculo (1-sim 2-nao)");
                calc=int.Parse(Console.ReadLine());
                
            } while (calc != 1 && calc != 2);
            
        } while (calc == 1);

    }
}

File: Desafios em JavaScript/1118 - Várias Notas Com Validação.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

var m, n, calc=1;

do
{
    do
    {
        m=parseFloat(lines.shift());

        if(m < 0.0 || m > 10.0)
        {
            console.log("nota invalida");
        }
    
    } while (m < 0.0 || m > 10.0);

    do
    {
        n=parseFloat(lines.shift());

        if(n < 0.0 || n > 10.0)
        {
            console.log("nota invalida");
        }

    } while (n < 0.0 || n > 10.0);

    let result = (m + n) / 2;
    console.log(`media = ${result.toFixed(2)}`);

    do
    {
        console.log("novo calculo (1-sim 2-nao)");
        calc=parseInt(lines.shift());
        
    } while (calc != 1 && calc != 2);
    
} while (calc == 1);